plugins/agent_bridge_mcp_client.go

```diff
@@ -0,0 +1,434 @@
+		choice := resp.Choices[0]
+		if choice.FinishReason == nil {
+			return "", fmt.Errorf("LLM returned a choice with no finish reason")
+		}
+
+		messages = append(messages, &azopenai.ChatRequestAssistantMessage{
+			ToolCalls: choice.Message.ToolCalls,
+		})
+
+		if *choice.FinishReason == azopenai.CompletionsFinishReasonStopped {
+			if choice.Message == nil || choice.Message.Content == nil {
+				logger.Errorf("[+] ERROR: no content in the response")
+				return "", fmt.Errorf("no content in the response")
+			}
+			logger.Infof("[+] Stop is detected, final response is (%v) in %v round", *choice.Message.Content, round)
+			return *choice.Message.Content, nil
+		}
+
+		for _, toolCall := range choice.Message.ToolCalls {
+			functionToolCall, ok := toolCall.(*azopenai.ChatCompletionsFunctionToolCall)
+			if !ok {
+				continue
+			}
+			result, err := callMCPTool(*functionToolCall.Function.Name, functionToolCall.Function.Arguments)
+			if err != nil {
+				logger.Errorf("[+] Failed to call tool (%s): %v", *functionToolCall.Function.Name, err)
+				messages = append(messages, &azopenai.ChatRequestToolMessage{
+					Content:    azopenai.NewChatRequestToolMessageContent("An error occurred while calling the tool"),
+					ToolCallID: functionToolCall.ID,
+				})
+				continue
+			}
+
+			messages = append(messages, &azopenai.ChatRequestToolMessage{
+				Content:    azopenai.NewChatRequestToolMessageContent(result),
+				ToolCallID: functionToolCall.ID,
+			})
+		}
+
+	}
+
+	// We reached the limit of rounds
+	return "", fmt.Errorf("reached the limit of rounds")
+}
+
+func getChatCompletionFunctionDefinition(tool mcp.Tool) (*azopenai.ChatCompletionsFunctionToolDefinitionFunction, error) {
+	jsonBytes, err := json.Marshal(tool.InputSchema)
+	if err != nil {
+		logger.Fatalf("[+] Failed to marshal parameters of tool (%v): err=%v", tool, err)
+	}
+
+	// Create the function definition
+	functionDefinition := &azopenai.ChatCompletionsFunctionToolDefinitionFunction{
+		Name:        &tool.Name,
+		Description: &tool.Description,
+		Parameters:  jsonBytes,
+	}
+	return functionDefinition, nil
+}
+
+func deinitMCPClient() {
+	for _, config := range mcpConfig {
+		if config.Client != nil {
+			config.Client.Close()
+			config.Client = nil
+		}
+	}
+}
+
+func initMCPClient() error {
+	for name, config := range mcpConfig {
+		if config.Client != nil {
+			continue // already initialized
+		}
+		logger.Debugf("[+] initMCPClient(%s) ...", name)
+
+		// If the env configuration value start with '$' (like
+		// 'GITHUB_PERSONAL_ACCESS_TOKEN=$GITHUB_PERSONAL_ACCESS_TOKEN'), let's
+		// take the value from the environment variable.
+		for index, env := range config.Env {
+			tokens := strings.SplitN(env, "=", 2)
+			if len(tokens) == 2 && len(tokens[1]) > 0 && tokens[1][0] == '$' {
+				envValue := os.Getenv(tokens[1][1:])
+				if envValue == "" {
+					logger.Errorf("Environment variable %s not found", tokens[1][1:])
+					return fmt.Errorf("MCP configuration error")
```

Comment:
What is better here? return an error or continue (in this case, we let the config unchanged, that is not so bad. After all, the user could to want to pass a parameter that start with '$')